src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -679,8 +679,32 @@ private StateEditor doTraverse(State s0, RoutingRequest options, TraverseMode tr
 
             // Apply turn restrictions
             if (options.arriveBy && !canTurnOnto(backPSE, s0, backMode)) {
-                return null;
+                // A turn restriction exists that forbids this turn with the current traverseMode.
+                // If using a bike, or micromobility, try again while walking
+                if (traverseMode == TraverseMode.BICYCLE || traverseMode == TraverseMode.MICROMOBILITY) {
+                    return doTraverse(s0, options.bikeWalkingOptions, TraverseMode.WALK);
+                }
+                // After trying again with switching to walking, the backMode will not have changed. Therefore, this
+                // conditional makes an assumption that a transition from non-walking to walking occurs at the very end
+                // of the last state. In this case it might be possible to make said turn if an immediate transition to
+                // walking is assumed at the very end of the previous state.
```

Comment:
I know this is a rather complicated thing to try to explain simply, but I wonder if there's a clearer way to state all of this. 

```
After a recursive traversal of this edge where the mode was switched to `WALK`, the backMode will not have changed. ...
```

I'm not sure what to say for the remaining sentences, but how can a conditional make an assumption? I feel like that language could be simplified for clarity.